src/v/cluster/rm_stm.cc

```diff
@@ -2823,91 +2827,99 @@ ss::future<stm_snapshot> rm_stm::take_snapshot() {
-            tx_snapshot_v3 tx_ss;
-            fill_snapshot_wo_seqs(tx_ss);
-            for (const auto& entry : _log_state.seq_table) {
-                tx_ss.seqs.push_back(entry.second.entry.copy());
-            }
-            tx_ss.offset = _insync_offset;
+                fut_serialize = reflection::async_adl<tx_snapshot>{}.to(
+                  tx_ss_buf, std::move(tx_ss));
+            } else if (version == tx_snapshot_v3::version) {
+                tx_snapshot_v3 tx_ss;
+                fill_snapshot_wo_seqs(tx_ss);
+                for (const auto& entry : _log_state.seq_table) {
+                    tx_ss.seqs.push_back(entry.second.entry.copy());
+                }
+                tx_ss.offset = _insync_offset;
 
-            for (const auto& entry : _log_state.current_txes) {
-                tx_ss.tx_seqs.push_back(tx_snapshot_v3::tx_seqs_snapshot{
-                  .pid = entry.first, .tx_seq = entry.second.tx_seq});
-            }
+                for (const auto& entry : _log_state.current_txes) {
+                    tx_ss.tx_seqs.push_back(tx_snapshot_v3::tx_seqs_snapshot{
+                      .pid = entry.first, .tx_seq = entry.second.tx_seq});
+                }
 
-            for (const auto& entry : _log_state.expiration) {
-                tx_ss.expiration.push_back(tx_snapshot::expiration_snapshot{
-                  .pid = entry.first, .timeout = entry.second.timeout});
-            }
+                for (const auto& entry : _log_state.expiration) {
+                    tx_ss.expiration.push_back(tx_snapshot::expiration_snapshot{
+                      .pid = entry.first, .timeout = entry.second.timeout});
+                }
 
-            reflection::adl<tx_snapshot_v3>{}.to(tx_ss_buf, std::move(tx_ss));
-        } else if (version == tx_snapshot_v2::version) {
-            tx_snapshot_v2 tx_ss;
-            fill_snapshot_wo_seqs(tx_ss);
-            for (const auto& entry : _log_state.seq_table) {
-                tx_ss.seqs.push_back(entry.second.entry.copy());
-            }
-            tx_ss.offset = _insync_offset;
-            reflection::adl<tx_snapshot_v2>{}.to(tx_ss_buf, std::move(tx_ss));
-        } else if (version == tx_snapshot_v1::version) {
-            tx_snapshot_v1 tx_ss;
-            fill_snapshot_wo_seqs(tx_ss);
-            for (const auto& it : _log_state.seq_table) {
-                auto& entry = it.second.entry;
-                seq_entry_v1 seqs;
-                seqs.pid = entry.pid;
-                seqs.seq = entry.seq;
-                try {
-                    seqs.last_offset = to_log_offset(entry.last_offset);
-                } catch (...) {
-                    // ignoring outside the translation range errors
-                    continue;
+                fut_serialize = reflection::async_adl<tx_snapshot_v3>{}.to(
+                  tx_ss_buf, std::move(tx_ss));
+            } else if (version == tx_snapshot_v2::version) {
+                tx_snapshot_v2 tx_ss;
```

Comment:
it'd be nice if the changes were separate from the reformatting

Reply:
:-/, I wanted to but I think my IDE reformatted it as soon I wrote the code and I forgot about. will do so in future changes.